include/zephyr/bluetooth/att.h

```diff
@@ -91,6 +91,12 @@ size_t bt_eatt_count(struct bt_conn *conn);
 
 #endif /* CONFIG_BT_EATT */
 
+enum bt_att_chan_option {
+	BT_ATT_CHAN_UNENHANCED = BIT(0),
+	BT_ATT_CHAN_ENHANCED = BIT(1),
+	BT_ATT_CHAN_ANY = BT_ATT_CHAN_UNENHANCED | BT_ATT_CHAN_ENHANCED,
```

Comment:
Hmm, while reading the test in https://github.com/zephyrproject-rtos/zephyr/pull/45381, this option is actually more disruptive that I though. 

Assuming that the majority of uses of the GATT parameters initially sets all values to 0, the default value for the options is actually invalid (i.e. 0). 

Instead of having
0x00 = invalid
0x01 = ATT only
0x02 = EATT only
0x03 = ATT/EATT

should it perhaps be
0x00 = ATT/EATT
0x01 = ATT only
0x02 = EATT only

In other words, if neither `BT_ATT_CHAN_UNENHANCED` or `BT_ATT_CHAN_ENHANCED`, it will preserve the default behavior of sending on "whatever free channel comes first", without changing the functionality of what this PR is at this point. 

So my suggestion is
```suggestion
	BT_ATT_CHAN_ANY = 0
	BT_ATT_CHAN_UNENHANCED_ONLY = BIT(0),
	BT_ATT_CHAN_ENHANCED_ONLY = BIT(1),
```
and then add a check to verify that both `BT_ATT_CHAN_UNENHANCED_ONLY` and `BT_ATT_CHAN_ENHANCED_ONLY` cannot be set at the same time.

Reply:
Actually `BT_ATT_CHAN_OPT_ANY` should perhaps be `BT_ATT_CHAN_OPT_NONE`

Reply:
I think I'd like to get input from more regarding this, i.e. if `0x00` should retain the current default behavior, or if `0x03` should.

Reply:
Another possibility is for `0x00` and `0x03` both to mean "any".
```c
enum bt_att_chan_option {
        BT_ATT_BACKCOMPAT = 0x00,
	BT_ATT_CHAN_UNENHANCED = BIT(0),
	BT_ATT_CHAN_ENHANCED = BIT(1),
	BT_ATT_CHAN_ANY = BT_ATT_CHAN_UNENHANCED | BT_ATT_CHAN_ENHANCED,
};
```

And update `BT_ATT_CHAN_OPTION` to something like
```c
#define BT_ATT_CHAN_OPTION(_params)  \
    ((_params)->chan_option ? (_params)->chan_option : BT_ATT_CHAN_ANY)
```
I am unsure whether I like it or not.

Reply:
I would rather introduce a breaking change than having a forever-living backcompat change in the API :D

Reply:
I think we should go with your option and have `0x0` be "any". The application should use the named enumerations anyway, only the implementation cares about how the bitfield is laid out.